nni/nas/oneshot/pytorch/strategy.py

```diff
@@ -144,3 +144,75 @@ class RandomOneShot(OneShotStrategy):
 
     def __init__(self, **kwargs):
         super().__init__(RandomSamplingLightningModule, **kwargs)
+
+    def attach_model(self, base_model: Union[Model, nn.Module]):
+        _reason = 'The reason might be that you have used the wrong execution engine. Try to set engine to `oneshot` and try again.'
+
+        if isinstance(base_model, Model):
+            if not isinstance(base_model.python_object, nn.Module):
+                raise TypeError('Model is not a nn.Module. ' + _reason)
+            py_model: nn.Module = base_model.python_object
+            if not isinstance(base_model.evaluator, Lightning):
+                raise TypeError('Evaluator needs to be a lightning evaluator to make one-shot strategy work.')
+            evaluator_module: LightningModule = base_model.evaluator.module
+            evaluator_module.running_mode = 'oneshot'
+            evaluator_module.set_model(py_model)
+        else:
+            from nni.retiarii.evaluator.pytorch.lightning import ClassificationModule
+            evaluator_module = ClassificationModule()
+            evaluator_module.running_mode = 'oneshot'
+            evaluator_module.set_model(base_model)
+        self.model: BaseOneShotLightningModule = self.oneshot_module(evaluator_module, **self.oneshot_kwargs)
+
+    def run(self, base_model: Model, applied_mutators):
+        # one-shot strategy doesn't use ``applied_mutators``
+        # but get the "mutators" on their own
+
+        _reason = 'The reason might be that you have used the wrong execution engine. Try to set engine to `oneshot` and try again.'
+
+        if applied_mutators:
+            raise ValueError('Mutator is not empty. ' + _reason)
+
+        if not isinstance(base_model.evaluator, Lightning):
+            raise TypeError('Evaluator needs to be a lightning evaluator to make one-shot strategy work.')
+
+        self.attach_model(base_model)
```

Comment:
What's the difference between this and super-class?

Maybe `attach_model` can be put into base class?

Reply:
yes